### ISSUE TITLE
EnumString Error "ambiguous_associated_items"  (Issue #57644)

### ISSUE THREAD

**Issue by JRAndreassen** [Peternator7/strum#194]
First off...

Love the crate, it makes my life a lot easier...
Just did a rustup update
```
   stable-x86_64-pc-windows-msvc unchanged - rustc 1.56.1 (59eed8a2a 2021-11-01)
  nightly-x86_64-pc-windows-msvc unchanged - rustc 1.58.0-nightly (cc946fcd3 2021-11-18)
```
I now get an error on the EnumString...
Did I miss something ?

Thanks
  JR

```rust
#[derive(Clone, Debug, PartialEq, PartialOrd, Serialize, Deserialize,
    Display, AsRefStr, IntoStaticStr, EnumString, EnumMessage)] // Panic: , EnumIter)]
pub enum DataStatus {
    /// Enum Entry Status DataStatus::Unknown
    #[serde(rename = "daq^status^Unknown", alias = "unknown", alias = "none")]
    Unknown,
    /// Status DataStatus::Normal
    #[serde(rename = "daq^status^Normal", alias = "normal", alias = "ok", alias = "nominal")]
    Normal,
    /// Status DataStatus::Notice
    #[serde(rename = "daq^status^Notice", alias = "notice", alias = "elevated")]
    Notice,
    /// Status DataStatus::Warning
    #[serde(rename = "daq^status^Warning", alias = "warning")]
    Warning,
    /// Status DataStatus::Error
    #[serde(rename = "daq^status^Error", alias = "Error", alias = "error")]
    Error,
    /// Status DataStatus::Critical
    #[serde(rename = "daq^status^Critical", alias = "critical")]
    Critical,
    /// Status DataStatus::Panic
    #[serde(rename = "daq^status^Fatal", alias = "fatal", alias = "termial", alias = "panic")]
    Fatal,
} // enum DataStatus

```
```
   --> C:\Development\Rust\stuff1\libs\stuff2\src\data\telemetry.rs:92:39
    |
92  |     Display, AsRefStr, IntoStaticStr, EnumString, EnumMessage)] // Panic: , EnumIter)]
    |                                       ^^^^^^^^^^
    |
    = note: `#[deny(ambiguous_associated_items)]` on by default
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #57644 <https://github.com/rust-lang/rust/issues/57644>
note: `Error` could refer to the variant defined here
   --> C:\Development\Rust\stuff1\libs\stuff2\src\data\telemetry.rs:108:5
    |
108 |     Error,
    |     ^^^^^
note: `Error` could also refer to the associated type defined here
   --> C:\Users\JR\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\convert\mod.rs:473:5
    |
473 |     type Error;
    |     ^^^^^^^^^^^
    = note: this error originates in the derive macro `EnumString` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by Peternator7** [comment]
Glad it helps you out :) The new version added a `TryFrom<str>` impl, but it can cause an ambiguity if you have an enum with an `Error` variant. it should be fixed by #193 and I'm working on getting a patch out today

**Comment by JRAndreassen** [comment]
Awesome... 
Thanks for the snappy reply :) ...

Sorry for the dup, I did see #193, which is strange, since it was the most recent one...
